### ISSUE TITLE
Bot not responding to other bots in slack.

### ISSUE THREAD

**Issue by Kcharle** [botman/driver-slack#6]
- BotMan Version: 1.5.7
- PHP Version: 7.1.7
- Messaging Service(s):
Slack
- Cache Driver:
Slack RTM Driver && Slack Rest Driver
### Description:

I have been working with the slack RTM driver over the last week in hopes to build a bot that will mainly interact with real users, but occasionally will interact with other bots I have running in my slack. The main feature I want to use bot to bot communication for is allowing my bot to self update. 

I have a server provisioned with Laravel Forge, and have a forge bot that posts to a channel in my slack with a success or failed deployment message. What I want to happen is for me to say "hey bot, freshen up", to which my bot would set a flag that it is deploying, make a request to forge to queue deployment, and once the deployment is done, my bot hears the forge bot say "Deployed some fresh code" and then clears the deploying flag and responds "Ok I'm feeling fresh!"

The problem is that my bot does not seem to pick up on listening to other bots, it works with users obviously, but not with other bots. I have tried playing around with the isBot logic in the RTM driver but no luck.

I also was working with Marcel a few weeks back to modify the isBot logic in the slack REST api driver so it could also listen for bots, but had no luck as well. Here is a forked repo where I tried to modify the REST api as well with no luck. https://github.com/charliebois/botman/tree/1.5

### Steps To Reproduce:

Provision server with forge.
Setup botman on RTM driver
Setup forge bot to message success to a slack channel
Add botman to the same channel as forge bot
Add a botman hears command to listen for "Username deployed some fresh code!"
Bot will not respond to other bot messages.

**Comment by Kcharle** [comment]
Example

When I send a message from my user
![image](https://user-images.githubusercontent.com/4724339/28724082-e33665fc-7386-11e7-926e-f51496b418a6.png)

The logs report properly , first line is my message saying not the bot, then botman responds, then it detects the next message from the bot is the botman Bot.
![image](https://user-images.githubusercontent.com/4724339/28724110-f90676ce-7386-11e7-84cc-e649cb5de06f.png)

But when I deploy to forge and the forge bot sends the message
![image](https://user-images.githubusercontent.com/4724339/28724198-3b777f8a-7387-11e7-9a6c-10d22cd03497.png)

And the log reports it is the bot as you can see on the last line
![image](https://user-images.githubusercontent.com/4724339/28724239-5995fbc2-7387-11e7-86a9-f0f82fa20744.png)

**Comment by Kcharle** [comment]
I'm still leaning towards the reason this is not working is because a bot has a *botID* and a *userID* and we are comparing the bots *userID* to the message event *botID*

![image](https://user-images.githubusercontent.com/4724339/28738785-905f2264-73c3-11e7-889a-28cb1ee07700.png)
The API call above hit's auth.test api endpoint which returns the userID which we are setting as the botID.

That's why I went down the road in my fork of doing getting the userID via the same auth.test API endpoint, but then hitting a second endpoint being the user *users.info* api endpoint which will return the bot's userID as well as botID. They do not mention it in the payload for that endpoint but I have tested and am able to get a userID and botID for a bot user. https://api.slack.com/methods/users.info
